yogstation/code/modules/donor/donor_verbs.dm

```diff
@@ -0,0 +1,143 @@
+
+var/list/donor_verbs_list = list(
+	/client/proc/donor_ooc
+	)
+
+/client/proc/add_donor_verbs()
+	if(is_donator(src))
+		verbs += donor_verbs_list
+
+/client/proc/remove_donor_verbs()
+	if(is_donator(src))
+		verbs.Remove(
+			donor_verbs_list,
+			)
+
+var/global/normal_donor_ooc_colour = "#333333"
+
+/client/proc/donor_ooc_admin(msg as text)
+	set name = "Donor OOC"
+	set category = "Admin"
+	donor_ooc(msg)
+
+/client/proc/donor_ooc(msg as text)
+	set name = "Donor OOC"
+	set category = "OOC"
+
+	if(GLOB.say_disabled)	//This is here to try to identify lag problems
+		to_chat(usr, "<span class='danger'>Speech is currently admin-disabled.</span>")
+		return
+
+	if(!mob)	return
+	if(IsGuestKey(key))
+		to_chat(src, "Guests may not use Donor.")
+		return
+
+	msg = copytext(sanitize(msg), 1, MAX_MESSAGE_LEN)
+	if(!msg)	return
+
+	if(!(prefs.toggles & CHAT_OOC))
+		to_chat(src, "<span class='danger'>You have Donor muted.</span>")
+		return
+
+	if(!holder)
+		if(!GLOB.ooc_allowed)
+			to_chat(src, "<span class='danger'>Donor is globally muted.</span>")
+			return
+		if(!GLOB.dooc_allowed && (mob.stat == DEAD))
+			to_chat(usr, "<span class='danger'>Donor for dead mobs has been turned off.</span>")
+			return
+		if(prefs.muted & MUTE_OOC)
+			to_chat(src, "<span class='danger'>You cannot use Donor (muted).</span>")
+			return
+		if(handle_spam_prevention(msg,MUTE_OOC))
+			return
+		if(findtext(msg, "byond://"))
+			to_chat(src, "<B>Advertising other servers is not allowed.</B>")
+			log_admin("[key_name(src)] has attempted to advertise in Donor: [msg]")
+			message_admins("[key_name_admin(src)] has attempted to advertise in Donor: [msg]")
+			return
+
```

Comment:
You're not sanitizing, limiting the length and applying the pretty filter to the message.